packages/maskbook/src/plugins/RedPacket/hooks/useAllRedPackets.ts

```diff
@@ -0,0 +1,9 @@
+import { useAsyncRetry } from 'react-use'
+import { RedPacketRPC } from '../messages'
+
+export function useAllRedPackets(address: string) {
+    return useAsyncRetry(async () => {
+        const payloads = await RedPacketRPC.getAllRedPackets(address)
```

Comment:
are you sure to design like this? it will be very expensive to transfer if the data amount is too large.

Reply:
Currently I think it is okay, ITO has the same design since there's no user send too many red packets or ITOs. We can check the data amount for each user on subgraph. So this will be redesigned later.